lib/Sema/TypeCheckDecl.cpp

```diff
@@ -5528,6 +5534,9 @@ void TypeChecker::addImplicitConstructors(NominalTypeDecl *decl) {
                         *this, classDecl, superclassCtor, kind)) {
         Context.addSynthesizedDecl(ctor);
         classDecl->addMember(ctor);
+
+        if (classDecl->hasValidatedLayout())
+          requestMemberLayout(ctor);
```

Comment:
This means every member of every class will have its layout computed -- this defeats the purpose of separating member layout from validateDecl(). I'm surprised we don't have a test for this :(